internal/resource/config/serversettings/server_settings_system_keys_resource.go

```diff
@@ -0,0 +1,396 @@
+
+	// Set refreshed state
+	diags = resp.State.Set(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+}
+
+// Update updates the resource and sets the updated Terraform state on success.
+func (r *serverSettingsSystemKeysResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	updateServerSettingsSystemKeys(ctx, req, resp, r.apiClient, r.providerConfig)
+}
+
+func updateServerSettingsSystemKeys(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse, apiClient *client.APIClient, providerConfig internaltypes.ProviderConfiguration) {
+	// Retrieve values from plan
+	var plan serverSettingsSystemKeysResourceModel
+
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	createServerSettingsSystemKeys := client.NewSystemKeysWithDefaults()
+	err := addServerSettingsSystemKeysFields(ctx, createServerSettingsSystemKeys, plan)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to add optional properties to add request for System Keys", err.Error())
+		return
+	}
+	requestJson, err := createServerSettingsSystemKeys.MarshalJSON()
+	if err == nil {
+		tflog.Debug(ctx, "Add request: "+string(requestJson))
+	}
+
+	apiCreateServerSettingsSystemKeys := apiClient.ServerSettingsApi.UpdateSystemKeys(config.ProviderBasicAuthContext(ctx, providerConfig))
+	apiCreateServerSettingsSystemKeys = apiCreateServerSettingsSystemKeys.Body(*createServerSettingsSystemKeys)
+	serverSettingsSystemKeysResponse, httpResp, err := apiClient.ServerSettingsApi.UpdateSystemKeysExecute(apiCreateServerSettingsSystemKeys)
+	if err != nil {
+		config.ReportHttpError(ctx, &resp.Diagnostics, "An error occurred while creating the ServerSettingsSystemKeys", err, httpResp)
+		return
+	}
+	responseJson, err := serverSettingsSystemKeysResponse.MarshalJSON()
+	if err == nil {
+		tflog.Debug(ctx, "Add response: "+string(responseJson))
+	}
+
+	// Read the response into the state
+	var state serverSettingsSystemKeysResourceModel
+
+	readServerSettingsSystemKeysResponse(ctx, serverSettingsSystemKeysResponse, &state)
+	diags = resp.State.Set(ctx, state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+}
+
+// This config object is edit-only, so Terraform can't delete it.
+func (r *serverSettingsSystemKeysResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+}
+
+func (r *serverSettingsSystemKeysResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	importServerSettingsSystemKeysLocation(ctx, req, resp)
+}
+func importServerSettingsSystemKeysLocation(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	// Retrieve import ID and save to id attribute
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
```

Comment:
Might be good to put a comment here saying that the id provided doesn't matter, since this appears to be a singleton resource